src/esp/metadata/attributes/AttributesBase.cpp

```diff
@@ -55,6 +55,36 @@ std::string getTranslationOriginName(int translationOrigin) {
   return "default";
 }
 
+AbstractAttributes::AbstractAttributes(const std::string& attributesClassKey,
+                                       const std::string& handle)
+    : Configuration() {
+  // set up an existing subgroup for user_defined attributes
+  addSubgroup("user_defined");
+  AbstractAttributes::setClassKey(attributesClassKey);
+  AbstractAttributes::setHandle(handle);
+  // set initial vals, will be overwritten when registered
+  set("ID", 0);
+  set("fileDirectory", "");
+}
+
+AbstractAttributes::AbstractAttributes(const AbstractAttributes& otr)
+    : Configuration(otr) {}
+AbstractAttributes::AbstractAttributes(AbstractAttributes&& otr) noexcept
+    : Configuration(std::move(otr)) {}
+
+AbstractAttributes& AbstractAttributes::operator=(
+    const AbstractAttributes& otr) {
+  if (this != &otr) {
+    this->Configuration::operator=(otr);
+  }
+  return *this;
+}
+AbstractAttributes& AbstractAttributes::operator=(
+    AbstractAttributes&& otr) noexcept {
+  this->Configuration::operator=(std::move(otr));
```

Comment:
```suggestion
  this->Configuration = std::move(otr);
```
isn't this equivalent?

Reply:
Don't think it works that way.  Configuration is a base class for the AbstractAttributes.

Reply:
Although that would be really cool, if you could assign to a base class component of a derived class that way.